src/main/java/hudson/security/LDAPSecurityRealm.java

```diff
@@ -970,9 +971,31 @@ public String getName() {
         @Override
         public GroupDetailsImpl mapAttributes(String dn, Attributes attributes) throws NamingException {
             LdapName name = new LdapName(dn);
-            String groupName = fixGroupname(String.valueOf(name.getRdn(name.size() - 1).getValue()));
+            String groupName = fixGroupname(extractGroupName(name, attributes));
             return new GroupDetailsImpl(dn, groupName);
         }
+
+        static String extractGroupName(LdapName name, Attributes attributes) throws NamingException {
+            final String CN = "cn";
+            boolean isCN = false;
+            String groupName = String.valueOf(name.getRdn(name.size() - 1).getValue());
+            for (NamingEnumeration ae = attributes.getAll(); ae.hasMore();) {
+                Attribute attr = (Attribute) ae.next();
+                if (CN.equals(attr.getID())) {
```

Comment:
attribute names are case insensitive.  The API does not mention that it lower cases them and this only matches the lower case.  Not sure about the interop here.